inc/admin-bar.php

```diff
@@ -127,3 +134,217 @@ function wp_head() {
 	<?php
 }
 add_action( 'wp_head', __NAMESPACE__ . '\wp_head' );
+
+/**
+ * Add `Irving` admin bar nodes. Provides shortcut links for the API, settings,
+ * and documentation.
+ *
+ * @param \WP_Admin_Bar $admin_bar WP Admin Bar object.
+ */
+function add_admin_bar_nodes( \WP_Admin_Bar $admin_bar ) {
+
+	if (
+		! current_user_can(
+			/**
+			 * Capability required to see the Irving admin bar.
+			 *
+			 * @param string Required. Capability.
+			 */
+			apply_filters( 'wp_irving_api_link_cap', 'manage_options' )
+		)
+	) {
+		return;
+	}
+
+	array_map(
+		[ $admin_bar, 'add_node' ],
+		array_merge(
+			[
+				// Base `Irving` node.
+				[
+					'id'    => 'irving',
+					'title' => __( 'Irving Development', 'wp-irving' ),
+					'href'  => home_url(),
+				],
+			],
+			get_api_items(), // Global API links.
+			get_admin_api_items(), // Admin-only API links.
+			get_frontend_api_items(), // Frontend-only API links.
+			get_setting_items(), // Irving settings.
+			get_documentation_items() // Documentation.
+		)
+	);
+}
+add_action( 'admin_bar_menu', __NAMESPACE__ . '\add_admin_bar_nodes', 90 );
+
+/**
+ * Get an array of Irving api node args to be passed into
+ * admin_bar->add_node().
+ *
+ * @return array
+ */
+function get_api_items(): array {
+	return [
+		[
+			'href'   => Components_Endpoint::get_wp_irving_api_url( '/' ),
+			'id'     => 'irving-api-links',
+			'parent' => 'irving',
+			'title'  => __( 'Components API', 'wp-irving' ),
+		],
+		[
+			'href'   => Components_Endpoint::get_wp_irving_api_url( '/' ),
+			'id'     => 'irving-homepage-api',
+			'parent' => 'irving-api-links',
+			'title'  => __( 'Homepage API', 'wp-irving' ),
+		],
+		[
+			'href'   => rest_url( 'irving/v1/registered-components/' ),
+			'id'     => 'irving-api-links-registered-components',
+			'parent' => 'irving-api-links',
+			'title'  => __( 'Registered Components API', 'wp-irving' ),
+		],
+	];
+}
+
+/**
+ * Get an array of Irving api node args to be passed into
+ * admin_bar->add_node().
+ *
+ * @return array
+ */
+function get_admin_api_items(): array {
+
+	if ( ! is_admin() ) {
+		return [];
+	}
+
+	$admin_api_items = [];
+
+	/**
+	 * Add a node for the post we're currently editing.
+	 */
+	$post_id = absint( $_GET['post'] ?? 0 ); // phpcs:ignore
```

Comment:
use the same phpcs disable syntax for consistency?

Reply:
I want to use `ignore` actually. Updating now. Thx.